### PR TITLE
Create clone method

### PR DESCRIPTION
**Pull request by ianwhitedeveloper**
Creating initial pull request for review. Please suggest improvements. Thanks :)

### CODE REVIEW

File: src/plugins/clone/clone.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        u(destination).on(type, events[type][i]);
+      }
+    }
+  }
+};
+
+u.prototype.getAll = function getAll (context, tag) {
+  // Mostly code borrowed from jQuery: https://github.com/jquery/jquery/blob/305f193aa57014dc7d8fa0739a3fefd47166cd44/src/manipulation.js
+  var rtn = u(tag || '*', context);
+  return rtn.length ? rtn.nodes : [context];
```
Comment (franciscop) [inline src/plugins/clone/clone.js]: Probably should be

```js
return rtn.lenght ? rtn.nodes.concat(context) : [context];
```

Which *might* be simplified to:

```js
return [context].concat(u(tag || '*', context).nodes);
```

**Comment by franciscop** [comment]
The point of the extensions was to let the programmer decide whether it should be a deep copy or a shallow one for cases such as `.append()`, which cannot really be decided if we set the options right into the `.clone(options)` method.

So, there are really two options:

1. All of the written extensions for `.mirror` are included without options. So the user can change them just by setting them to `false`. This would effectively change the options for all of the `.clone()` methods.
1. Add a third parameter for `.after()` where we pass the options and they get passed to `.clone()`.

I think that `1.` is better to avoid clutter. IMO doing it by default (and allowing a global option for nulling them or adding more) is:

1. Way better than jquery.
1. Better than cluttering the number of parameters, the documentation and the tests.
1. Maintaining them is much easier in the long run (also 2. helps for maintainability).
1. `.clone()` has a much more internal usage in Umbrella than in jquery, since `.append()` automatically clones them so the developer doesn't need to do it

**Comment by ianwhitedeveloper** [comment]
@franciscop  got it. making changes now. Thanks for the feedback!

**Comment by franciscop** [comment]
Now it looks great so I'll merge it. As always, beware [prenature optimization](http://stackoverflow.com/a/9592751/938236) that makes the code more complex.

**Pull request merged**